### PR TITLE
EIP 3540 Addition: EOF1 contracts can only DELEGATECALL EOF1 contracts

### PR DESCRIPTION
**Pull request by ritzdorf**
As suggested by Paweł Bylica (@chfast) I am proposing this EIP addition.

Full discussion here: https://ethereum-magicians.org/t/evm-object-format-eof/5727/48

### CODE REVIEW

File: EIPS/eip-3540.md
Comment (gumb0) [inline EIPS/eip-3540.md:167]: Maybe makes sense to mention that target address still becomes warm (is added to `accessed_addresses`, if not already there) even if it doesn't contain EOF1 code

Comment (ritzdorf) [reply EIPS/eip-3540.md:167]: Thanks @gumb0 , excellent suggestion. Will add that.

**Comment by SamWilsn** [comment]
Since there were commits after the previous approval, this'll require a new approval from one of @axic, @chfast, or @gumb0.

File: EIPS/eip-3540.md
Comment (chfast) [inline EIPS/eip-3540.md:167]:
```suggestion
7. When an EOF1 contract performs a `DELEGATECALL` the target must be EOF1. If it is not EOF1, the `DELEGATECALL` exceptionally halts. Hence, (among other things) all the gas passed to the `DELEGATECALL` is consumed and 0 is pushed onto the stack. `DELEGATECALL` from EOF1 to an empty container also fails. In case the `DELEGATECALL` fails as described here, the target address still becomes warm.
```

**Review by chfast — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
